test/e2e/test/step.go

```diff
@@ -42,11 +49,18 @@ func (l StepList) RunSequential(t *testing.T) {
 			continue
 		}
 		if !t.Run(ts.Name, ts.Test) {
-			logf.Log.Error(errors.New("test failure"), "stopping early")
+			logf.Log.Error(errors.New("test failure"), "continuing with additional tests")
 			if ts.OnFailure != nil {
 				ts.OnFailure()
 			}
-			t.FailNow()
```

Comment:
What does it mean for nightly e2e tests where `JOB_NAME` is not defined? Shouldn't we continue to `t.FailNow()` for them?

Reply:
https://github.com/elastic/cloud-on-k8s/pull/6363/files#diff-e9a5bfc47dbabe806572ddf5571346ffce93e837d675487f39da1a828fc74555R51
```
  JOB_NAME=${BUILDKITE_PIPELINE_NAME}
```

Am I misunderstanding something? Isn't `JOB_NAME` always going to be set, whether in jenkins, or in BK?

Reply:
I misread and thought it was only set for e2e/main to start. So we are enabling this for all e2e-tests for all providers.

Don't we need to update [pipeline-nightly-e2e-tests.yml](https://github.com/elastic/cloud-on-k8s/blob/2bf050d8fe19921f2bc770a1bb3840f9eb6acf33/.buildkite/pipeline-nightly-e2e-tests.yml) like for [pipeline.yml](https://github.com/elastic/cloud-on-k8s/pull/6363/files#diff-a3991ebf1475eb82acab13946ffc1eca02b43917f6d5bec3898f45c8b0b9bd53) where we split the cluster creation from the execution of the tests? I think https://github.com/elastic/cloud-on-k8s/pull/6525 would really help to do that.

Reply:
Ok, I think I've got this configured for the nightly bits. Thanks for that. I'll work to trigger that to fully test it as soon as I can.

Reply:
> Don't we need to update [pipeline-nightly-e2e-tests.yml](https://github.com/elastic/cloud-on-k8s/blob/2bf050d8fe19921f2bc770a1bb3840f9eb6acf33/.buildkite/pipeline-nightly-e2e-tests.yml) like for [pipeline.yml](https://github.com/elastic/cloud-on-k8s/pull/6363/files#diff-a3991ebf1475eb82acab13946ffc1eca02b43917f6d5bec3898f45c8b0b9bd53) where we split the cluster creation from the execution of the tests? I think https://github.com/elastic/cloud-on-k8s/pull/6525 would really help to do that.

The splitting up of the pipelines is an optimisation that happened to be made in this PR but is actually not essential to the feature we are implementing here. But of course we can add it for the nightly pipeline as well for consistency.